File: LeetCode题目集/14. Longest Common Prefix(easy)-水平扫描法.cpp
/*
 @Date    : 2019-01-06 22:47:42
 @Author  : 酸饺子 (changzheng300@foxmail.com)
 @Link    : https://github.com/SourDumplings
 @Version : $Id$
*/

/*
https://leetcode.com/problems/longest-common-prefix/
 */

class Solution
{
public:
    string longestCommonPrefix(vector<string>& strs)
    {
        if (strs.empty())
        {
            return "";
        }
        string prefix = strs[0];
        for (int i = 1; i < strs.size(); ++i)
        {
            while (strs[i].find(prefix) != 0)
            {
                prefix = prefix.substr(0, prefix.length() - 1);
            }
            if (prefix.empty())
            {
                return "";
            }
        }
        return prefix;
    }
};

File: OJ practices/LeetCode题目集/14. Longest Common Prefix(easy)-水平扫描法.java
/**
 * @Date    : 2019-01-06 22:35:54
 * @Author  : 酸饺子 (changzheng300@foxmail.com)
 * @Link    : https://github.com/SourDumplings
 * @Version : $Id$
 *
 * https://leetcode.com/problems/longest-common-prefix/
*/

class Solution
{
    public String longestCommonPrefix(String[] strs)
    {
        if (strs.length == 0)
            return "";
        String prefix = strs[0];
        for (int i = 1; i != strs.length; ++i)
        {
            while (strs[i].indexOf(prefix) != 0)
            {
                prefix = prefix.substring(0, prefix.length() - 1);
            }
            if (prefix.isEmpty())
                return "";
        }
        return prefix;
    }
}

